src/librbd/ObjectMap.h

```diff
@@ -69,6 +69,11 @@ class ObjectMap {
                   const ZTracer::Trace &parent_trace, T *callback_object) {
     assert(start_object_no < end_object_no);
     if (snap_id == CEPH_NOSNAP) {
+      end_object_no = std::min(end_object_no, m_object_map.size());
```

Comment:
Can you make this change here [1] as well?

[1] https://github.com/trociny/ceph/blob/acd0c1d397219a93445007f60ab1e6b492dfd42d/src/librbd/ObjectMap.cc#L305

Reply:
@dillaman done